mex/backend/identity/main.py

```diff
@@ -24,6 +29,17 @@ def assign_identity(request: IdentityAssignRequest) -> Identity:
         raise MExError("found multiple identities indicating graph inconsistency")
     if len(graph_result.data) == 1:
         identity = transform_identity_result_to_identity(graph_result.data[0])
+    elif (
+        request.identifierInPrimarySource
+        == MEX_PRIMARY_SOURCE_IDENTIFIER_IN_PRIMARY_SOURCE
+        and request.hadPrimarySource == MEX_PRIMARY_SOURCE_STABLE_TARGET_ID
```

Comment:
I don't understand this case, could you please add a comment on this? If I understand correctly, you are comparing to two constants here. I thought we only have the cases of: 
1. ID already in graph -> return that
2. ID not yet in graph -> generate new one

What is this case covering?